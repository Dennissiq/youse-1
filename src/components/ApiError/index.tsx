import React from 'react'

export default ({ text }: { text: string }) => <div>{text}</div>
