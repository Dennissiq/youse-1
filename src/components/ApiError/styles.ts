import styled from 'styled-components'
import { yellow } from '../../constants'

export const ApiError = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`

export const SVGWarning = styled.svg`
  height: 60px;
  fill: ${yellow};
  margin-bottom: 12px;
`
