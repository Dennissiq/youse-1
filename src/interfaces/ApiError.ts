export default interface ApiError {
  code: number
  tryAgain: (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>
  ) => void | undefined
  page: number
  filter: string
  isLoading: boolean
  userIsTyping: boolean
}
