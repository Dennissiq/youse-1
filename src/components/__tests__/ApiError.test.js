import React from 'react'
import renderer from 'react-test-renderer'
import ApiError from '../ApiError'

it('renders correctly', () => {
  const tree = renderer.create(<ApiError text="Error" />).toJSON()
  expect(tree).toMatchSnapshot()
})
