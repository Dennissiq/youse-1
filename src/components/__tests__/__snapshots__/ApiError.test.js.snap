// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdnylx fnOHxw"
>
  <svg
    className="sc-gtssRu jnwQvU"
    viewBox="0 0 320.83 320.83"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M21.617 290.246h277.604c9.469 0 17.013-4.553 20.168-12.184 2.904-7.011 1.36-14.8-4.107-21.212L180.497 39.684l-1.333-1.333c-5.009-5.009-11.667-7.767-18.748-7.767-7.082 0-13.739 2.758-18.743 7.767l-1.333 1.333L5.55 256.844c-5.466 6.413-7.016 14.207-4.106 21.212 3.159 7.632 10.698 12.19 20.173 12.19zM160.416 69.213L277.35 257.606H43.487L160.416 69.213z"
    />
    <path
      d="M160.416 105.345c-9.197 0-16.654 7.457-16.654 16.654v71.388c0 9.197 7.457 16.654 16.654 16.654s16.654-7.457 16.654-16.654v-71.388c.006-9.197-7.451-16.654-16.654-16.654z"
    />
    <circle
      cx="160.248"
      cy="234.903"
      r="16.486"
    />
  </svg>
  Error
</div>
`;
